wandb/sdk/wandb_run.py

```diff
@@ -2428,10 +2428,6 @@ def _console_stop(self) -> None:
             self._output_writer.close()
             self._output_writer = None
 
-    def _on_init(self) -> None:
```

Comment:
what do you think of keeping the _on_xyz pattern but just move some of the code that now in in the init function in wandb_init to on_init?

Reply:
I don't think we should have this pattern. There are parts of `wandb_init` that should be the responsibility of a `Run`, but `_on_init` is the wrong approach.

Reply:
why? i think it is nice that there is clearly defined function on_init, on_start etc...

Reply:
the name should describe what it does rather than when it runs

Reply:
if you look at other python libraries there many cases when this patter is used, so not sure it is actually a valid reason, since it is fair to say that this is the action we do on init, this on start and this on finish for example, i don't think it is necessarily wrong, but in any case we should have a uniform pattern in anyone we choose

Reply:
a lot has been written about this topic; e.g. https://nikas.praninskas.com/javascript/2016/11/28/naming-things-handlers/